### PR TITLE
Cross compile to Scala 3

### PR DESCRIPTION
**Pull request by tiagomota**
Fix https://github.com/SwissBorg/pekko-persistence-postgres/issues/6
Depends on #26

### CODE REVIEW

File: .github/workflows/scala.yml [hunk 1]
```diff
@@ -11,10 +11,11 @@ on:
 jobs:
   test:
     runs-on: ubuntu-latest
-    name: Java ${{matrix.java}}
+    name: Test with Scala ${{ matrix.scala }} and Java ${{ matrix.java }})
     strategy:
       matrix:
         java: ['temurin:1.11', 'temurin:1.17', 'temurin:1.21']
+        scala: [ 2.13.13, 3.3.1 ]
```
Comment (lomigmegard) [inline .github/workflows/scala.yml:18]: We should use latest version from LTS, which is [3.3.3](https://github.com/scala/scala3/releases/tag/3.3.3)

File: core/src/main/scala/org/apache/pekko/persistence/postgres/snapshot/PostgresSnapshotStore.scala [hunk 2]
```diff
@@ -65,7 +65,6 @@ class PostgresSnapshotStore(config: Config) extends SnapshotStore {
         snapshotDao.snapshotForMaxSequenceNr(persistenceId, maxSequenceNr)
       case SnapshotSelectionCriteria(maxSequenceNr, maxTimestamp, _, _) =>
         snapshotDao.snapshotForMaxSequenceNrAndMaxTimestamp(persistenceId, maxSequenceNr, maxTimestamp)
-      case _ => Future.successful(None)
```
Comment (lomigmegard) [inline core/src/main/scala/org/apache/pekko/persistence/postgres/snapshot/PostgresSnapshotStore.scala:68]: question: unreachable code?

File: project/Dependencies.scala [hunk 3]
```diff
@@ -2,7 +2,8 @@ import sbt.*
 
 object Dependencies {
   val Scala213 = "2.13.13"
-  val ScalaVersions = Seq(Scala213)
+  val Scala3 = "3.3.1"
```
Comment (lomigmegard) [inline project/Dependencies.scala:5]: 3.3.3

File: core/src/main/scala/org/apache/pekko/persistence/postgres/snapshot/PostgresSnapshotStore.scala (hunk 2 above)
Comment (tiagomota) [reply core/src/main/scala/org/apache/pekko/persistence/postgres/snapshot/PostgresSnapshotStore.scala:68]: 🎯 Exactly!

File: core/src/main/scala/org/apache/pekko/persistence/postgres/db/ExtendedPostgresProfile.scala [hunk 4]
```diff
@@ -13,7 +13,6 @@ trait ExtendedPostgresProfile
     with PgSearchSupport
     with PgNetSupport
     with PgLTreeSupport
-    with PgJsonSupport
```
Comment (lomigmegard) [inline core/src/main/scala/org/apache/pekko/persistence/postgres/db/ExtendedPostgresProfile.scala:16]: looks ok to me, the only place where we use Json column is for metadata, and we rely on `io.circe.Json`.

**Review by lomigmegard — APPROVED** [review]

**Comment by lomigmegard** [comment]
suggestion: in `ExtendedPostgresProfile`, the type could be `MyAPI` no? It's strange to use the object's type and shouldn't change the implicits resolution.

```
  override val api: MyAPI.type = MyAPI
```

**Pull request merged**